[#1003] Update code architecture doc. for CSB part (#1004)
kiesraad/e-ks: docs/code-architecture.md

Reviewer: ```suggestion
2. **Tracing and security headers.** HTTP tracing is opened, and the security
response headers (CSP, `X-Frame-Options`, etc.) are scheduled for development.
```
Perhaps link an issue too

for example `candidate_lists/importer.rs` (CSV/EML import) and
`political_groups/steps.rs` (multi-step flow state).

### The CSB section (`src/csb/`)

`src/csb/` holds the central voting bureau side of the application: CSB
members import the packages submitted by political groups and examine them
(Hoofdstuk I of the Kieswet). It mirrors the `src/pg/` conventions (the same
`pages/`, `forms/`, `extractors/`, `structs/`, `components/` layout, with
`CsbContext` in place of `Context`), but its access model is fundamentally
different: a political group only ever sees its own stream, while a committee
member works across all imported streams.

#### Scopes

> Sessions and streams both carry a `Scope` (`src/core/scope.rs`):

Reviewer: ```suggestion
Sessions and streams both carry a `Scope` (`src/core/scope.rs`) of one of the following variants:
```

  (scope `ImportedByCsb`), driven by `CsbEvent`. The projection holds the
  imported snapshot (`imported_data`), a second projection with the paper
  corrections replayed on top (`paper_corrected_data`), the recorded
  omissions and person corrections, and the examination-finished flag.
- **`CsbMainStoreData`** (`src/csb/store_main/`), a single stream per
  election shared by all committee members under the fixed
  `CSB_MAIN_STREAM_ID` (scope `CentralElectoralCommittee`). It records
  committee-wide events (currently logins) and backs the main CSB audit log.

#### Domains

- **`index`**: the CSB home page.
- **`import`**: brings a submitted package into the CSB side. The documents
  generated on the PG side embed the chain hash of the event they were
  rendered from; a committee member enters that hash (a unique prefix
  suffices) and the import locates the matching event
  (`find_event_by_hash_prefix`, backed by the `events_hash_idx` index),
  replays the source stream up to it (`PgStoreData::snapshot_until`), and
  persists the snapshot as a `CsbEvent::Import` on a **fresh** `ImportedByCsb`
  stream. The political group's own stream is never written to, and importing

>   the same source stream twice is rejected.

Reviewer: ```suggestion
  the same source stream twice is rejected (might change with #999).
```

- **`examination`**: the examination of the imported lists. An overview
  groups the imported political groups by finished/unfinished; detail pages
  render the imported data read-only; omissions and corrections are recorded
  in overlays; and the model **I 4** notice (the letter listing every
  recoverable omission across all imported streams, per electoral district)
  is generated here.
- **`monitoring`**: an overview of the political-group streams built from
  `StreamMeta`: event counts and timestamps read from the backend's index.
  This deliberately reads no payloads: no stream key is unwrapped and nothing
  is decrypted, so monitoring works without touching any political group's
  data.
- **`audit_log`**: the CSB audit log, a read view over either the main
  committee stream or a single imported stream.

#### Omissions and corrections

An **omission** (*verzuim*) is a defect found during examination.
`OmissionCategory` ties each omission to what it concerns: the political
group itself, a candidate list (with the affected electoral districts), or a
candidate (with the affected lists). Recoverable omissions feed the I 4

> notice. A **correction** (`CsbEvent::UpdateCorrection`) records a fix to

Reviewer: ```suggestion
notice. A **correction** (*ambtshalve correctie*) (`CsbEvent::UpdateCorrection`) records a fix to
```

> imported person data (initials, last name, date of birth, place of

Reviewer: > imported person data

It also includes corrections on the appellation

> residence); corrected persons are kept in a separate map in the projection

Reviewer: ```suggestion
residence); corrections on persons are kept in a separate map in the projection
```
(this is more accurate if/when #1002 gets merged)
